src/persist/src/mem.rs

```diff
@@ -582,9 +583,46 @@ impl MemMultiRegistry {
     }
 }
 
+#[derive(Debug)]
+struct MemConsensus {
+    data: Arc<Mutex<Option<VersionedData>>>,
+}
+
+impl Default for MemConsensus {
+    fn default() -> Self {
+        Self {
+            data: Arc::new(Mutex::new(None)),
+        }
+    }
+}
+
+#[async_trait]
+impl Consensus for MemConsensus {
+    async fn head(&self, _deadline: Instant) -> Result<Option<VersionedData>, Error> {
+        Ok(self.data.lock()?.clone())
+    }
+
+    async fn compare_and_swap(
+        &mut self,
+        expected: Option<SeqNo>,
+        new: Option<VersionedData>,
+        _deadline: Instant,
+    ) -> Result<Result<(), Option<VersionedData>>, Error> {
+        let mut data = self.data.lock()?;
+        let seqno = data.as_ref().map(|data| data.seqno);
+        if seqno != expected {
+            return Ok(Err(data.clone()));
+        }
+
+        *data = new;
```

Comment:
hmm, this totally losing SeqNo information on a "deletion", such that SeqNo can regress by going through a deletion has me worried. I think I'm going to change my mind here and say that `new` shouldn't be an option and we should leave deletions entirely as a TODO until we figure out more precisely what we need there

Reply:
done!